### ISSUE TITLE
Flaky test: UnitsNet.Tests.UnitSystemTests.PositiveInfinityFormatting

### ISSUE THREAD

**Issue by angularsen** [angularsen/UnitsNet#436]
This only happens very rarely on AppVeyor (first time I can think of), but @gojanpaolo reports it happens all the time on his setup. I assume this is due to manipulating static values in other tests, such as:

```
GetDefaultAbbreviationFallsBackToUsEnglishCulture()
{
    var zuluCulture = new CultureInfo("zu-ZA");
    CultureInfo.CurrentCulture = CultureInfo.CurrentUICulture = zuluCulture;
```

https://ci.appveyor.com/project/angularsen/unitsnet/build/724/tests

`UnitsNet.Tests.UnitSystemTests.PositiveInfinityFormatting`
```
Assert.Equal() Failure
          ↓ (pos 0)
Expected: ∞ m
Actual:   Infinity m
          ↑ (pos 0)
```

The following:
```
Console.WriteLine(double.PositiveInfinity.ToString(CultureInfo.InvariantCulture));
Console.WriteLine(double.PositiveInfinity.ToString(new CultureInfo("zu-ZA")));

/* Prints:
Infinity
∞
*/
```

So the fix is likely one or both of:
1) Ensure tests that rely on manipulating CultureInfo or other static values are NOT run in parallel (can xunit selectively configure this?)
2) Set CultureInfo properties back to their original values so it doesn't pollute other tests in same test class (I think all tests in one class are run in the same app domain and share static values)

**Comment by gojanpaolo** [comment]
I think #1 is possible by applying [test collection definition](https://github.com/xunit/xunit/issues/1179#issuecomment-339015963)

I'll try #2 also then submit a PR whichever fixes the issue.

**Comment by gojanpaolo** [comment]
I think fix#1 is possible by applying [test collection definition](https://github.com/xunit/xunit/issues/1179#issuecomment-339015963)

I'll try fix#2 also then submit a PR whichever fixes the issue.

**Comment by angularsen** [comment]
It would be hard to verify.

You are also right that on computers with a different culture, this test may always fail since it doesn't try to specify a culture. That test should specify InvariantCulture, which is a third fix on this problem - but it only fixes this particular test.

I would do all 3 really, since they all seem like the right thing to do in my mind. Then I also wouldn't care too much to try to verify if it actually fixes it, let's just assume it does until it proves us wrong down the road ;)
